fpdf/fpdf.py

```diff
@@ -1765,49 +1770,100 @@ def add_font(self, family, style="", fname=None, uni="DEPRECATED"):
         if fontkey in self.fonts or fontkey in self.core_fonts:
             warnings.warn(f"Core font or font already added '{fontkey}': doing nothing")
             return
+
         for parent in (".", FPDF_FONT_DIR):
             if not parent:
                 continue
+
             if (Path(parent) / fname).exists():
                 ttffilename = Path(parent) / fname
                 break
         else:
             raise FileNotFoundError(f"TTF Font file not found: {fname}")
 
-        # include numbers in the subset! (if alias present)
-        # ensure that alias is mapped 1-by-1 additionally (must be replaceable)
-        sbarr = "\x00 "
-        if self.str_alias_nb_pages:
-            sbarr += "0123456789"
-            sbarr += self.str_alias_nb_pages
+        # font tools
+        ft = ttLib.TTFont(ttffilename)
+
+        scale = 1000 / ft["head"].unitsPerEm
+        ascent = ft["hhea"].ascent * scale
+        descent = ft["hhea"].descent * scale
+        try:
+            capHeight = ft["OS/2"].sCapHeight * scale
+        except AttributeError:
+            capHeight = ascent
+        bbox = (
+            f"[{ft['head'].xMin * scale:.0f} {ft['head'].yMin * scale:.0f}"
+            f" {ft['head'].xMax * scale:.0f} {ft['head'].yMax * scale:.0f}]"
+        )
+        stemV = 50 + int(pow((ft["OS/2"].usWeightClass / 65), 2))
+        italicAngle = ft["post"].italicAngle
+        underlinePosition = ft["post"].underlinePosition * scale
+        underlineThickness = ft["post"].underlineThickness * scale
+
+        flags = 4
```

Comment:
Could you maybe add a comment explaining what are those flags?

Reply:
Maybe use an Enum?

Reply:
For now, I don't know what are those flags :grin: these are parts of `ttfonts.py` that I extracted to get the exact same data as before.
I see that they are [used](https://github.com/PyFPDF/fpdf2/blob/6f8c326c36cdd653d75ef537ace7efd693205f21/fpdf/fpdf.py#L4157) with some bit manipulation to write something in the PDF, maybe I can find some information on the PDF reference manual...

Reply:
I changed the `flags` integer in an Enum with docstrings explaining the various flags